### ISSUE TITLE
UniRep or BERT fine tuning examples?

### ISSUE THREAD

**Issue by KirkDCO** [songlab-cal/tape#71]
I'm just beginning to work with the TAPE package, specifically with the hope of testing both UniRep and BERT protein embeddings in my system.  I anticipate that I will need to do some fine tuning of the base models first, but I'm a bit unsure how to approach that.  Are there any examples of this approach for either of these models?

**Comment by rmrao** [comment]
[This section of the README](https://github.com/songlab-cal/tape#training-a-downstream-model) describes how to finetune a model - is there anything more specific you would need? 

As an FYI, pytorch does not support mLSTMs particularly well, so UniRep training in pytorch is quite slow and memory intensive. This seems to be a pretty difficult limitation to work around without delving into manually implementing CUDA extensions, unfortunately.

**Comment by KirkDCO** [comment]
Thank you for the reply.

Looking at that section, it is titled "Training a Downstream Model" which I interpret as training a "top" model.  For example, in image recognition if I took the VGG16 model and removed the final dense layers, I could generate my own top model using the outputs of the VGG16 convolutional blocks, but those convolutional blocks would not themselves be trained.  Rather, if I started with the trained VGG16 model but allowed the convolutional layers to be trainable, I could run a few epochs of fine tuning.

For the protein models, I interpreted Training a Downstream Model as maintaining the early layers of the network and only training the later, top, or downstream layers.  For my project, I would like to be able to fine tune all layers of the network to my target protein class.  In the UniRep paper, they refer to this as EvoTuning, I believe.

Or, I have completely misunderstood that section.  8^)
